service/resolver.go

```diff
@@ -119,40 +119,64 @@ func (r *resolve) MapK8sResourceAthenzResource(k8sRes string) string {
 	return athenzRes
 }
 
-// createAthenzDomain use cfg.ServiceAthenzDomain;
+// createAthenzDomains use cfg.ServiceAthenzDomains;
 
+// do the following for each cfg.ServiceAthenzDomains
 // split it with ".";
 // for each token, if it match /^_.*_$/ but not "_namespace_", replace the token with config.GetActualValue(token);
 // and then return the processed value
-func (r *resolve) createAthenzDomain() string {
-	reps := make([]string, 0, strings.Count(r.cfg.ServiceAthenzDomain, ".")+1)
-	for _, v := range strings.Split(r.cfg.ServiceAthenzDomain, ".") {
-		if v != "_namespace_" && strings.HasPrefix(v, "_") && strings.HasSuffix(v, "_") {
-			// Note: If deploying in a different namespace than the kube-public namespace, change it to get information from kube api
-			reps = append(reps, v, config.GetActualValue(v))
+func (r *resolve) createAthenzDomains() []string {
+	domains := make([]string, 0, len(r.cfg.ServiceAthenzDomains))
+	// Default domain string length is 128. so, maxi domain depth is 64.
+	// If this size is exceeded, it is a slice, so there is no problem.
+	// https://github.com/yahoo/athenz/blob/b92bd87bf65cf605a1abc2c6263974d85b4bb5d1/servers/zms/conf/zms.properties#L269
+	reps := make([]string, 0, 64)
+	for _, domain := range r.cfg.ServiceAthenzDomains {
+		reps = reps[:0]
+		for _, v := range strings.Split(domain, ".") {
+			if v != "_namespace_" && strings.HasPrefix(v, "_") && strings.HasSuffix(v, "_") {
+				// Note: If deploying in a different namespace than the kube-public namespace, change it to get information from kube api
+				reps = append(reps, v, config.GetActualValue(v))
+			}
 		}
+		domains = append(domains, strings.NewReplacer(reps...).Replace(domain))
 	}
-	return strings.NewReplacer(reps...).Replace(r.cfg.ServiceAthenzDomain)
+	return domains
+}
+
+// BuildDomainsFromNamespace returns domains by processing athenzDomains.
```

Comment:
suggest to copy `buildAthenzDomain` function's comment to this function

Reply:
thank you. fixed 3b3e61e